WooCommerce/Classes/System/SessionManager.swift

```diff
@@ -159,11 +160,28 @@ final class SessionManager: SessionManagerProtocol {
     /// Nukes all of the known Session's properties.
     ///
     func reset() {
+        deleteApplicationPassword()
         defaultAccount = nil
         defaultCredentials = nil
         defaultStoreID = nil
         defaultSite = nil
     }
+
+    /// Deletes application password
+    ///
+    func deleteApplicationPassword() {
+        guard case let .wporg(username, password, siteAddress) = loadCredentials(),
+              let usecase = try? DefaultApplicationPasswordUseCase(username: username,
+                                                                   password: password,
+                                                                   siteAddress: siteAddress,
+                                                                   keychain: keychain) else {
+            return
+        }
+
+        Task {
+            try await usecase.deletePassword()
```

Comment:
Since `usecase` is a local variable, is it possible that it's deallocated before the request is completed? Should we keep a strong reference to it in the scope of the class to avoid that?

Reply:
> A task runs regardless of whether you keep a reference to it. However, if you discard the reference to a task, you give up the ability to wait for that task’s result or cancel the task.

https://developer.apple.com/documentation/swift/task

I believe the unstructured `Task` will take care of keeping a reference to the `usecase` and make sure that the task is finished. 

Please correct me if I am wrong.

Reply:
My understanding of the quoted documentation is that the wrapping `Task` will not need to be held on to for reference. I'm concerned about the `usecase` though, whether the API request made by `deletePassword` can be finished if `usecase` is deallocated? I haven't checked but maybe there's one way to test this - we can make `usecase` a variable and set it to nil below the `Task` closure.

Reply:
If I try to make sendable a `var` instead of `let`, I cannot pass it to the unstructured `Task`. I get the following error. 
```
Reference to captured var 'usecase' in concurrently-executing code
```

My understanding is,
1. I can only pass a `let` `usecase ` into the task because it is a [`Sendable Types`](https://docs.swift.org/swift-book/LanguageGuide/Concurrency.html#ID649)
2. `DefaultApplicationPasswordUseCase` is a sendable type even though it is a `class`, because it doesn't have any mutable state. If we introduce a `var` property in it, we can no longer pass a reference of `DefaultApplicationPasswordUseCase` to the unstructured `Task`
3. The above two points make it clear that the unstructured `Task` captures a reference of `DefaultApplicationPasswordUseCase` until it finishes. 
4. `SessionManager` is a singleton and we can be sure that it will not be deallocated before the `Task` finishes.  

I believe this makes it clear that we don't have to hold a reference to the `usecase`.

Kindly let me know your thoughts.

Reply:
Thanks for the explanation! It makes sense to me, so please go ahead with the current implementation 👍